deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java

```diff
@@ -457,13 +466,38 @@ private void emitMetrics(AgentBean updateBean) {
                                     env.getEnv_name(),
                                     env.getStage_name()),
                             "success",
-                            String.valueOf(updateBean.getStatus().equals(AgentStatus.SUCCEEDED)))
+                            String.valueOf(AgentStatus.SUCCEEDED.equals(updateBean.getStatus())))
                     .increment();
+
+            Long hostStartTime = estimateHostStartTime(updateBean);
+            if (hostStartTime != null) {
+                Metrics.timer(
+                                String.format(
+                                        LAUNCH_LATENCY_TIMER_NAME,
+                                        env.getEnv_name(),
+                                        env.getStage_name()))
+                        .record(
+                                Duration.ofMillis(
+                                        updateBean.getFirst_deploy_time() - hostStartTime));
+            }
         } catch (Exception ex) {
             LOG.warn("Failed to emit metrics of {}", updateBean, ex);
         }
     }
 
+    private Long estimateHostStartTime(AgentBean agent) {
+        try {
+            List<HostBean> hosts = hostDAO.getHostsByHostId(agent.getHost_id());
+            if (!hosts.isEmpty()) {
+                return hosts.get(0).getCreate_date();
+            }
+        } catch (Exception ex) {
+            LOG.info("Failed to get host with id {}", agent.getHost_id(), ex);
```

Comment:
Could be a warn similar to [previous change](https://github.com/pinterest/teletraan/blob/master/deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java#L242) so that the level matches, but may not matter much since it seems unexpected

Reply:
Good catch.